### PR TITLE
Add autocomplete and clarify type parameters for text selector

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

- Documents newly supported `autocomplete` parameter for text selector
- Alters documentation of `type` parameter to better clarify usage consistent with `autocomplete`
- Some spelling fixes
- Some fixes for bad "is will" grammar

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#81060, home-assistant/frontend#14202
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by bdraco — DISMISSED** [review]

File: source/_docs/blueprint/selectors.markdown [hunk 1]
```diff
@@ -18,24 +18,29 @@ The following selectors are currently available:
 - [Action selector](#action-selector)
 - [Add-on selector](#add-on-selector)
 - [Area selector](#area-selector)
+  - [Example area selectors](#example-area-selectors)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown]: This is not a TOC. Please revert these changes.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/blueprint/selectors.markdown (hunk 1 above)
Comment (steverep) [reply source/_docs/blueprint/selectors.markdown]: Done.  I didn't even realize it did that.  The `markdown.extension.toc.updateOnSave` setting forced it.

Comment (frenck) [reply source/_docs/blueprint/selectors.markdown]: Yeah, I figured, that setting can be annoying at times

**Review by frenck — APPROVED** [review]
Thanks, @steverep 👍
